core/src/ibc/connection_03/manager.rs

```diff
@@ -55,6 +60,105 @@ impl Manager {
+        desired_identifier: Identifier,
+        counterparty_connection_identifier: Identifier,
+        counterparty_prefix: CommitmentPrefix,
+        counterparty_client_identifier: Identifier,
+        client_identifier: Identifier,
+        proof_init: CommitmentProof,
+        proof_consensus: CommitmentProof,
+        proof_height: u64,
+        consensus_height: u64,
+    ) -> Result<(), String> {
+        let current_height = ctx.get_current_height();
+        if consensus_height > current_height {
+            return Err(format!(
+                "Consensus height {} is greater than current height {}",
+                consensus_height, current_height
+            ))
+        }
+        let expected = ConnectionEnd {
+            state: ConnectionState::INIT,
+            counterparty_connection_identifier: desired_identifier.clone(),
+            counterparty_prefix: get_commiment_prefix(),
+            client_identifier: counterparty_client_identifier.clone(),
+            counterparty_client_identifier: client_identifier.clone(),
+        };
+
+        let connection = ConnectionEnd {
+            state: ConnectionState::TRYOPEN,
+            counterparty_connection_identifier: counterparty_client_identifier.clone(),
+            counterparty_prefix: counterparty_prefix.clone(),
+            client_identifier: client_identifier.clone(),
+            counterparty_client_identifier: counterparty_client_identifier.clone(),
+        };
+
+        self.verify_connection_state(ctx, &connection, proof_height, proof_init, desired_identifier.clone(), &expected);
+
+        if let Some(previous_connection_end) = self.query(ctx, &desired_identifier) {
+            let expected_init = ConnectionEnd {
+                state: ConnectionState::INIT,
+                counterparty_connection_identifier,
+                counterparty_prefix,
+                client_identifier,
+                counterparty_client_identifier,
+            };
+            if previous_connection_end != expected_init {
+                return Err(format!(
+                    "Invalid previous connection status: previous: {:?}, expected: {:?}",
+                    previous_connection_end, expected_init
+                ))
+            }
+        }
+
+        let kv_store = ctx.get_kv_store();
+        kv_store.set(&connection_path(&desired_identifier), &connection.rlp_bytes());
+        Ok(())
+    }
+
+    fn query(&self, ctx: &mut dyn ibc::Context, identifier: &str) -> Option<ConnectionEnd> {
+        let kv_store = ctx.get_kv_store();
+
+        let path = connection_path(&identifier);
+        if kv_store.has(&path) {
+            let raw = kv_store.get(&path);
+            let connection_end = rlp::decode(&raw).expect("Only the connection code can save the code");
+            return Some(connection_end)
+        }
+
+        None
+    }
+
+    fn verify_connection_state(
+        &self,
+        ctx: &mut dyn ibc::Context,
+        connection: &ConnectionEnd,
```

Comment:
Could you explain the meaning of this argument?

Reply:
I checked the spec. The spec is not using it. Please ignore the argument.